qcore/helpers.py

```diff
@@ -258,21 +258,32 @@ def copy_public_attrs(source_obj, dest_obj):
             setattr(dest_obj, name, value)
 
 
-def object_from_string(st):
+def object_from_string(name):
     """Creates a Python class or function from its fully qualified name.
 
-    :param st: A fully qualified name of a class or a function.
+    :param name: A fully qualified name of a class or a function. In Python 3 this
+        is only allowed to be of text type (unicode). In Python 2, both bytes and unicode
+        are allowed.
     :return: A function or class object.
 
     This method is used by serialization code to create a function or class
     from a fully qualified name.
 
     """
-    pos = st.rfind('.')
+    if six.PY3:
+        if not isinstance(name, six.text_type):
```

Comment:
Since this is within an `if six.PY3` branch, you don't really need to use six, you can just use `str` here (and `(str, unicode)` below in the py2 branch).